File: internal/models/customer_card.go
package models

type CustomerCardCreate struct {
	Surname     *string
	Name        *string
	Patronymic  *string
	PhoneNumber *string
	City        *string
	Street      *string
	ZipCode     *string
	Percent     *int
}

type CustomerCardRetrieve struct {
	CardNumber  *string
	Surname     *string
	Name        *string
	Patronymic  *string
	PhoneNumber *string
	City        *string
	Street      *string
	ZipCode     *string
	Percent     *int
}

type CustomerCardUpdate struct {
	Surname     *string
	Name        *string
	Patronymic  *string
	PhoneNumber *string
	City        *string
	Street      *string
	ZipCode     *string
	Percent     *int
}

File: frontend/src/types/customer_card.ts
export interface CustomerCard {
  card_number: string;
  cust_surname: string;
  cust_name: string;
  cust_patronymic?: string;
  phone_number: string;
  city?: string;
  street?: string;
  zip_code?: string;
  percent: number;
}

export type CustomerCardCreate = Omit<CustomerCard, 'card_number'>;
export type CustomerCardUpdate = Partial<CustomerCardCreate>; 